### PR TITLE
latex2html: add "test do" block

### PR DESCRIPTION
**Pull request by razah-dev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Adding a ``` test do ``` block in the formula ```latex2html```

### CODE REVIEW

**Comment by bfontaine** [comment]
Can’t we write a test that does something more substantial than e.g. printing the version or the help?

**Comment by fxcoudert** [comment]
Idea for a test: create a file `test.tex` with one line `\textbf{it works!}`.
Run `latex2html test.tex`.
Check that this creates a file named `test/test.html` which contains `it works!`.

**Comment by raza15** [comment]
@fxcoudert already done that. Will be pushing changes in a while

**Comment by raza15** [comment]
@bfontaine @fxcoudert added non-trivial tests too. Let me know if you like it!

File: Formula/latex2html.rb [hunk 1]
```diff
@@ -20,4 +22,43 @@ def install
                           "--with-texpath=#{share}/texmf/tex/latex/html"
     system "make", "install"
   end
+
+  test do
+    # Trivial Tests
+    assert_match version.to_s, shell_output("#{bin}/latex2html --version")
+    system "#{bin}/latex2html", "--help"
```
Comment (bfontaine) [inline Formula/latex2html.rb]: You can remove those now that there’s a non-trivial test below.

File: Formula/latex2html.rb [hunk 2]
```diff
@@ -20,4 +22,43 @@ def install
...
+
+    # Non-Trivial Test
```
Comment (bfontaine) [inline Formula/latex2html.rb]: Please remove this comment.

File: Formula/latex2html.rb [hunk 3]
```diff
@@ -20,4 +22,43 @@ def install
...
+      \begin{document}
+
+      \maketitle
+
+      \section{Experimental Setup}
+        \textbf{it works!}
+      \end{document}
+    '
+    test_file.puts(contents_of_file)
+    test_file.close
```
Comment (bfontaine) [inline Formula/latex2html.rb]: You can create a file with the following code:
```ruby
(testpath/"test.tex").write <<-EOS.undent
  \documentclass{article}
  ...
EOS

File: Formula/latex2html.rb [hunk 4]
```diff
@@ -20,4 +22,43 @@ def install
...
+
+    system "#{bin}/latex2html", "test.tex"
+
+    # Check if the 'section' links exist in the main html file
+    main_html_file = File.read("test/test.html")
+    assert_match /Experimental Setup/, main_html_file
+    # Now, goes to that html file and checks the content
```
Comment (bfontaine) [inline Formula/latex2html.rb]: You can’t assume `nokogiri` is installed. The `assert_match` above is sufficient; thanks!

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/latex2html.rb (hunk 3 above)
Comment (raza15) [inline Formula/latex2html.rb]: when i was creating a file with this code, it was giving error that ``` invalid unicode escape \usepackage ```. Do u want me to do it anyway?

Comment (bfontaine) [inline Formula/latex2html.rb]: You’ll have to double every backslack like so:
(testpath/"test.tex").write <<-EOS.undent
  \\documentclass{article}
  \\usepackage{...}
  ...
EOS

Comment (raza15) [inline Formula/latex2html.rb]: Great. Thanks!

**Comment by raza15** [comment]
@bfontaine made all the changes you requested.

**Comment by fxcoudert** [comment]
Thanks @raza15 for your contribution to Homebrew!

**Pull request merged**